lib/core/test/unit/Cardano/Wallet/DB/Arbitrary.hs

```diff
@@ -160,155 +159,166 @@ import qualified Data.ByteString as BS
+arbitraryBlockchainParameters :: BlockchainParameters
+arbitraryBlockchainParameters =
+    genesisParameters { getEpochStability = Quantity maxBound }
 
-instance Arbitrary (RndState DummyTarget) where
-    shrink (RndState k ix addrs pending g) =
-        [ RndState k ix' addrs' pending' g
-        | (ix', addrs', pending') <- shrink (ix, addrs, pending) ]
-    arbitrary = RndState
+instance GenState s => Arbitrary (Wallet s DummyTarget) where
+    shrink w =
+        [ unsafeInitWallet u (currentTip w) s (blockchainParameters w)
+        | (u, s) <- shrink (utxo w, getState w) ]
+    arbitrary = unsafeInitWallet
         <$> arbitrary
         <*> arbitrary
         <*> arbitrary
-        <*> arbitrary
-        <*> (mkStdGen <$> arbitrary)
+        <*> pure arbitraryBlockchainParameters
 
-instance Arbitrary (Index 'Hardened 'AccountK) where
+instance Arbitrary (PrimaryKey WalletId) where
     shrink _ = []
-    arbitrary = arbitraryBoundedEnum
+    arbitrary = do
+        bytes <- B8.pack . pure <$> elements ['a'..'k']
+        return $ PrimaryKey $ WalletId $ hash bytes
 
-instance Arbitrary (Index 'Hardened 'AddressK) where
+instance Arbitrary WalletMetadata where
     shrink _ = []
-    arbitrary = arbitraryBoundedEnum
-
-instance Arbitrary DummyTarget.Tx where
-    shrink (DummyTarget.Tx ins outs) =
-        [DummyTarget.Tx ins' outs | ins' <- shrinkList' ins ] ++
-        [DummyTarget.Tx ins outs' | outs' <- shrinkList' outs ]
+    arbitrary =  WalletMetadata
+        <$> (WalletName <$> elements ["bulbazaur", "charmander", "squirtle"])
+        <*> genUniformTime
+        <*> (fmap WalletPassphraseInfo <$> liftArbitrary genUniformTime)
+        <*> oneof [pure Ready, Restoring . Quantity <$> genPercentage]
+        <*> pure NotDelegating
       where
-        shrinkList' xs  = filter (not . null)
-            [ take n xs | Positive n <- shrink (Positive $ length xs) ]
-
-    arbitrary = Tx
-        <$> fmap (L.nub . L.take 5 . getNonEmpty) arbitrary
-        <*> fmap (L.take 5 . getNonEmpty) arbitrary
+        genPercentage :: Gen Percentage
+        genPercentage = do
+            let (Right upperBound) = mkPercentage @Int 100
+            arbitraryBoundedEnum `suchThat` (/= upperBound)
 
-instance Arbitrary TxMeta where
-    shrink _ = []
-    arbitrary = TxMeta
-        <$> arbitrary
-        <*> elements [Incoming, Outgoing]
-        <*> arbitrary
-        <*> fmap (Quantity . fromIntegral) (arbitrary @Word32)
+{-------------------------------------------------------------------------------
+                                   Blocks
+-------------------------------------------------------------------------------}
 
-instance Arbitrary TxStatus where
-    arbitrary =
-        elements [Pending, InLedger, Invalidated]
+instance Arbitrary BlockHeader where
+    arbitrary = do
+        sid@(SlotId ep sl) <- arbitrary
+        let h = fromIntegral sl * fromIntegral len + fromIntegral ep * 100
```

Comment:
why 100? maybe it is good to name it ?

Reply:
Indeed, it's the same constant as for the generated slot id (which is arbitrary). I'll use a named constant :+1: